plugins/techdocs-module-mkdocs/src/components/MkDocsContentTransformers/Code/Code.tsx

```diff
@@ -72,26 +74,37 @@ const CopyToClipboardButton = ({ text }: CopyToClipboardButtonProps) => {
   );
 };
 
-import type { Transformer } from './transformer';
+const CODE_SELECTOR = 'pre > code';
 
 /**
  * Recreates copy-to-clipboard functionality attached to <code> snippets that
  * is native to mkdocs-material theme.
  */
-export const copyToClipboard = (theme: Theme): Transformer => {
-  return dom => {
-    const codes = dom.querySelectorAll('pre > code');
-    for (const code of codes) {
-      const text = code.textContent || '';
-      const container = document.createElement('div');
-      code?.parentElement?.prepend(container);
-      ReactDom.render(
-        <ThemeProvider theme={theme}>
-          <CopyToClipboardButton text={text} />
-        </ThemeProvider>,
-        container,
-      );
-    }
-    return dom;
-  };
+export const CodeTransformer = () => {
+  const dom = useTechDocsShadowDom();
+  const theme = useTheme();
+
+  if (!dom) return null;
+
+  const codes = dom.querySelectorAll(CODE_SELECTOR);
+
+  if (!codes.length) return null;
+
+  return (
+    <>
+      {Array.from(codes).map((code, index) => {
+        const text = code.textContent || '';
+        const container = document.createElement('div');
+        code.parentElement?.prepend(container);
+        return (
+          <Portal key={index} container={container}>
+            <ThemeProvider theme={theme}>
+              <CopyToClipboardButton text={text} />
+            </ThemeProvider>
+            ,
```

Comment:
not sure if its this causing this:

<img width="852" alt="Screenshot 2022-03-24 at 11 10 17" src="https://user-images.githubusercontent.com/25153114/159893617-178a8f0a-9445-4774-8d8c-90237df38ce2.png">